filter.cc

```diff
@@ -781,6 +798,15 @@ uint32_t Filter::serialize(char *buffer) {
   }
   totalSize += 1;
 
+  if (ruleDefinition) {
```

Comment:
I don't think this is serializing correctly, I think we would need to store the `ruleDefinitionSize` in the above `snprintf`

Reply:
Pls also add a test to `test/js/serializationTest.js` for that to ensure it works.